### ISSUE TITLE
Pasting from the clipboard inserts characters before the selection, unlike ordinary paste

### ISSUE THREAD

**Issue by donaldcallen** [mawww/kakoune#4914]
### Version of Kakoune

v2022.10.31

### Reproducer

Copy some text to the clipboard. Paste in a kakoune buffer with the cursor in the midst of some existing text.

### Outcome

Clipboard text is pasted before the cursor, like shift-p

### Expectations

I would have expected  <space>-p to behave like p, not P.

### Additional information

FreeBSD 13.2
sway (Wayland compositor)

**Comment by Screwtapello** [comment]
When you say "clipboard", do you mean Kakoune's `"` register (the default register for `y` and `p`) or the system clipboard?

How are you pasting, if not with `p` or `P`? Is there some other tool involved, like `xsel` or your terminal emulator's paste command?

**Comment by donaldcallen** [comment]
> When you say "clipboard", do you mean Kakoune's `"` register (the default register for `y` and `p`) or the system clipboard?
> 
> How are you pasting, if not with `p` or `P`? Is there some other tool involved, like `xsel` or your terminal emulator's paste command?

Apologies. This was my error. I am experimenting with several different editors and I managed to confuse myself.

**Comment by krobelus** [comment]
No I think you were right. When using the terminal emulator's paste shortcut we insert before and not after the selection.
As mentioned in b2cf74bb4 (Implement bracketed paste, 2022-12-11) this was only because I was used to that behavior from my existing mappings. Probably it makes sense to match p here, not P.

**Comment by donaldcallen** [comment]
On Fri, Jun 2, 2023 at 5:44 AM Johannes Altmanninger <
***@***.***> wrote:

> No I think you were right. When using the terminal emulator's paste
> shortcut we insert before and not after the selection.
>
I don't think I've ever been in a discussion before where I had to insist
that I was wrong :-)

What you say about the terminal emulator may be true, but I wasn't using
it. I had created my own mappings for yanking to and pasting from the
clipboard. But in switching
among several editors, such as Helix, where these commands are built in, I
completely forgot that I had added these mappings myself to kakoune. And
the paste mapping was wrong, inserting
before the selection rather than after. So the error was mine, easily fixed.
